libfwupdplugin/fu-string.c

```diff
@@ -439,6 +440,63 @@ fu_strsplit(const gchar *str, gsize sz, const gchar *delimiter, gint max_tokens)
 	return g_strsplit(str, delimiter, max_tokens);
 }
 
+/**
+ * fu_strsplit_stream:
+ * @stream: a #GInputStream to split
+ * @offset: offset into @stream
+ * @delimiter: a string which specifies the places at which to split the string
+ * @callback: (scope call) (closure user_data): a #FuStrsplitFunc.
+ * @user_data: user data
+ * @error: (nullable): optional return location for an error
+ *
+ * Splits the string, calling the given function for each
+ * of the tokens found. If any @callback returns %FALSE scanning is aborted.
+ *
+ * Use this function in preference to fu_strsplit() when the input file is untrusted,
+ * and you don't want to allocate a GStrv with billions of one byte items.
+ *
+ * Returns: %TRUE if no @callback returned FALSE
+ *
+ * Since: 1.8.2
```

Comment:
200

Reply:
Ack, fixed, thanks. Good catch, the PR is massive. I'm thinking we should merge it after the Algoltex one goes in.

Reply:
Something else I think you should do is mark the updated version of libjcat in subprojects and meson build.

Reply:
I need to release them first :)

Reply:
Okay, done. Phew!